docs/trestle_author_jinja.md

```diff
@@ -0,0 +1,110 @@
+# Trestle author jinja - output templating support for oscal documents.
+
+Unfortunately OSCAL documents are not yet universally accepted. Therefore to support various OSCAL and non-OSCAL compliance workflows `trestle author jinja` is designed to provide end users with the ability to use jinja to produce customized output. This complements the more structured commands `trestle author catalog-{assemble|generate}`, `trestle author profile-{assemble|generate}` and `trestle author ssp-{assemble|generate}` and allows arbitrary use of jinja.
+
+## Jinja and jinja extensions provided by trestle.
+
+[Jinja](https://jinja.palletsprojects.com/en/3.0.x/) is a powerful templating engine that is both more flexible that pure 'Moustache' approaches, and not coupled to a particular web application server. (as an example Django templates). Users are encouraged to review the [template designer documentation](https://jinja.palletsprojects.com/en/3.0.x/templates/) for jinja as all core functionality is exposed.
+
+Trestle's implementation of the Jinja command works in the following way:
+
+1. The template search space, by default, is relative to the current working directory.
+1. Trestle can inject a lookup table, into the jinja variables to contain booleans / substitutions required by end users using Moustache style variable substitutions.
+1. Trestle can provide a number of interfaces to OSCAL objects, currently a resolved catalog and a SSP, into jinja.
+1. Trestle supports custom jinja tags for importing.
+1. The jinja templating is recursive, to ensure all jinja tags are resolved as appropriate.
+
+More details will be on each of these points below.
+
+## CLI invocation
+
+Note the examples here use markdown, however, jinja can quite easily target xml or html if used w/o specific markdown content.
+`trestle author jinja -i input_template.md.jinja -o output_file.md -ssp SSP_NAME -p PROFILE_NAME -lut lookup_table.yaml -elp lut.prefix`
+
+- `-i` input file path, relative to cwd. Users are encouraged to use the `file_name.target_extension.jinja` best practice as it helps mitigate issues, however is not required.
+- `-o` final output path, relative to cwd.
+- `-ssp` (optional) ssp name (in the trestle project). When used a interface will be made available to the jinja template.
+- `-p` (optional) profile name (in the trestle project). When used a interface to the resolved catalog will be made available to the jinja template.
```

Comment:
I think from the end-user perspective it is a bit unclear what will be printed if those flags are passed. While the sections below provide some insights into what will be done if those flags are passed, but not sure if it is sufficient. Maybe adding a section like `if you pass -ssp then the following structure will be followed for each control...`

Reply:
I've changed this. I've also added a section for sample jinja templates.

Reply:
Great thanks will give it a read once pushed